### DISCUSSION TITLE
Collabra container does not start and massive log spam. Error shown is below.

### DISCUSSION THREAD

**Discussion by antonical** [nextcloud/all-in-one#2314] [Questions] [answered]
### Steps to reproduce
1. Run Docker AIO instance (domain is registered and DNS configured correctly.
2.
3.

### Expected behavior 
Collabra container should start
### Actual behavior 
massive log spam but at the bottom I get this.
Failed to listen on Server port(s) (9980-9980). Exiting.
wsd-00001-00001 2023-04-09 15:32:10.044687 +0100 [ coolwsd ] FTL  Failed to listen on Server port(s) (9980-9980). Exiting.| wsd/COOLWSD.cpp:5420
Forced Exit with code: 70
wsd-00001-00001 2023-04-09 15:32:10.044701 +0100 [ coolwsd ] FTL  Forced Exit with code: 70| common/Util.cpp:1118
cp: cannot create regular file '/opt/cool/systemplate/etc/resolv.conf': Permission denied
cp: cannot create regular file '/opt/cool/systemplate/etc/hosts': Permission denied

The ports are not firewalled off.

### Host OS 
Ubuntu 22.04

#### Nextcloud AIO version 
4.7.0
#### Current channel 
latest
#### Other valuable info 
curl -vvv https://$NC_DOMAIN:443/hosting/discovery
*   Trying x.x.x.x:443...
* Connected to mydomain.cloud (x.x.x.x) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=mydomain.cloud
*  start date: Apr  9 13:45:35 2023 GMT
*  expire date: Jul  8 13:45:34 2023 GMT
*  subjectAltName: host "mydomain.cloud" matched cert's "mydomain.cloud"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* h2h3 [:method: GET]
* h2h3 [:path: /hosting/discovery]
* h2h3 [:scheme: https]
* h2h3 [:authority: mydomain.cloud]
* h2h3 [user-agent: curl/7.83.1]
* h2h3 [accept: */*]
* Using Stream ID: 1 (easy handle 0x7f309eae01b0)
> GET /hosting/discovery HTTP/2
> Host: mydomain.cloud
> user-agent: curl/7.83.1
> accept: */*
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* Connection state changed (MAX_CONCURRENT_STREAMS == 250)!
< HTTP/2 502 
< alt-svc: h3=":443"; ma=2592000
< server: Caddy
< content-length: 0
< date: Sun, 09 Apr 2023 19:11:58 GMT
< 
* Connection #0 to host mydomain.cloud left intact

Cheers
Spart

**Comment by szaimen** [comment]
Looks like https://github.com/CollaboraOnline/online/issues/3102

**Comment by antonical** [comment]
This fix did not work:
This is the error on restarting.
Failed to listen on Server port(s) (9980-9980). Exiting.
wsd-00001-00001 2023-04-10 10:46:06.902652 +0100 [ coolwsd ] FTL  Failed to listen on Server port(s) (9980-9980). Exiting.| wsd/COOLWSD.cpp:5420
Forced Exit with code: 70
wsd-00001-00001 2023-04-10 10:46:06.902838 +0100 [ coolwsd ] FTL  Forced Exit with code: 70| common/Util.cpp:1118
cp: cannot create regular file '/opt/cool/systemplate/etc/resolv.conf': Permission denied
cp: cannot create regular file '/opt/cool/systemplate/etc/hosts': Permission denied
.
.
.
Failed to listen on Server port(s) (9980-9980). Exiting.
wsd-00001-00001 2023-04-10 10:49:38.903816 +0100 [ coolwsd ] FTL  Failed to listen on Server port(s) (9980-9980). Exiting.| wsd/COOLWSD.cpp:5420
Forced Exit with code: 70
wsd-00001-00001 2023-04-10 10:49:38.903993 +0100 [ coolwsd ] FTL  Forced Exit with code: 70| common/Util.cpp:1118
.
.
.
wsd-00001-00001 2023-04-10 10:56:58.329610 +0100 [ coolwsd ] DBG  Enabling commenting on [] extension files| wsd/COOLWSD.cpp:5048
wsd-00001-00001 2023-04-10 10:56:58.336383 +0100 [ coolwsd ] TRC  #-1: Created socket. Thread affinity set to 0x7f87e5e9cec0| net/Socket.hpp:379
wsd-00001-00001 2023-04-10 10:56:58.336415 +0100 [ coolwsd ] ERR  #-1: Failed set ipv6 socket to 0 (EBADF: Bad file descriptor)| net/Socket.cpp:791
wsd-00001-00001 2023-04-10 10:56:58.336499 +0100 [ coolwsd ] ERR  #-1: Failed to bind to: IPv6 port: 9980 (EBADF: Bad file descriptor)| net/Socket.cpp:798
wsd-00001-00001 2023-04-10 10:56:58.336514 +0100 [ coolwsd ] TRC  #-1: Socket dtor| net/Socket.hpp:146
wsd-00001-00001 2023-04-10 10:56:58.336523 +0100 [ coolwsd ] DBG  #-1: Closed socket to []| net/Socket.hpp:151
Failed to listen on Server port(s) (9980-9980). Exiting.
wsd-00001-00001 2023-04-10 10:56:58.336548 +0100 [ coolwsd ] FTL  Failed to listen on Server port(s) (9980-9980). Exiting.| wsd/COOLWSD.cpp:5420
Forced Exit with code: 70
wsd-00001-00001 2023-04-10 10:56:58.336562 +0100 [ coolwsd ] FTL  Forced Exit with code: 70| common/Util.cpp:1118
cp: cannot create regular file '/opt/cool/systemplate/etc/resolv.conf': Permission denied
cp: cannot create regular file '/opt/cool/systemplate/etc/hosts': Permission denied

Looks like it is still trying to use IPV6.

Start command is this:

sudo docker run --sig-proxy=false --name nextcloud-aio-mastercontainer --restart always --publish 80:80 --publish 8080:8080 --publish 8443:8443 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro -e SKIP_DOMAIN_VALIDATION=true -e "extra_params=--o:net.proto=IPv4" nextcloud/all-in-one:latest

**Comment by szaimen** [comment]
>  -e "extra_params=--o:net.proto=IPv4" nextcloud/all-in-one:latest

You cannot pass the value like this. The correct fix would be enabling ipv6 in your kernel.

**Reply by antonical** [reply]
That was the suggested fix in the post you pointed to.

We cannot enable IPV6 as it is not supported on our network. Why assume IPV6 and not IPV4 which would be the default on most servers.

Cheers

**Comment by szaimen** [comment]
You will also run into other issues when not enabling ipv6 that is why enabling ipv6 in the kernel is the better fix.

**Comment by szaimen** [comment]
And just as a sidenote: the bug is in collabora and not in aio, so should rather be fixed on the collabora side.

**Comment by szaimen** [comment]
I've found this for you: https://cloudzy.com/knowledge-base/configure-ipv6-on-ubuntu/#How_to_Set_Up_IPV6_on_Ubuntu

**Reply by antonical** [reply]
You seem to be missing the point. We know how to config IPV6 but it is not supported on our network so we will not be enabling it.

We are trialing the AIO and apart from this issue it seems to work well and is easy to admin.

How do we resolve this, It is a core component of the AIO for groupware applications.

Is there a collaboration (excuse the pun) between NC and Collabra, such that you are sharing issues and bugs?

For now we have stopped the containers and disabled it.

Cheers

**Comment by szaimen** [comment, marked as answer]
> We know how to config IPV6 but it is not supported on our network so we will not be enabling it.

Then you will not be able to use collabora until https://github.com/CollaboraOnline/online/issues/3102 is fixed.

**Reply by antonical** [reply]
Ok thanks. And are you aware if this issue is being fixed? Are Collabora aware of it and understand it?

Cheers

**Reply by nz2o** [reply]
Edit - I see what you're saying - the lack of ipv6 is what's causing me.... all kinds of grief?
